influxql/query_manager.go

```diff
@@ -222,10 +222,12 @@ func (qm *defaultQueryManager) Query(qid uint64) (*queryTask, bool) {
 	return query, ok
 }
 
-func (qm *defaultQueryManager) waitForQuery(qid uint64, timeout time.Duration, closing <-chan struct{}, monitorCh <-chan error) {
+func (qm *defaultQueryManager) waitForQuery(qid uint64, interrupt <-chan struct{}, timeout time.Duration, closing <-chan struct{}, monitorCh <-chan error) {
 	var timer <-chan time.Time
```

Comment:
We can talk about this for the next version, but I think this might be more trouble than it's worth. There would need to be a separate timer for each query that gets run and the pool would need to be able to dynamically grow and shrink with time. If the timer creation is very expensive it might be worth it, but I don't know if the bookkeeping needed to make a set of pooled timers work would be practical.